example/fine_advection/advection_package.cpp

```diff
@@ -101,35 +125,42 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
 }
 
 AmrTag CheckRefinement(MeshBlockData<Real> *rc) {
-  // refine on advected, for example.  could also be a derived quantity
-  static auto desc = parthenon::MakePackDescriptor<Conserved::phi>(rc);
-  auto pack = desc.GetPack(rc);
+  std::shared_ptr<StateDescriptor> pkg =
+      rc->GetMeshPointer()->packages.Get("advection_package");
+  auto do_regular_advection = pkg->Param<bool>("do_regular_advection");
+  if (do_regular_advection) {
+    // refine on advected, for example.  could also be a derived quantity
+    static auto desc = parthenon::MakePackDescriptor<Conserved::phi>(rc);
+    auto pack = desc.GetPack(rc);
 
-  auto pmb = rc->GetBlockPointer();
-  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
-  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
-  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+    auto pmb = rc->GetBlockPointer();
+    IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
+    IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
+    IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
 
-  typename Kokkos::MinMax<Real>::value_type minmax;
-  parthenon::par_reduce(
-      parthenon::loop_pattern_mdrange_tag, PARTHENON_AUTO_LABEL, DevExecSpace(),
-      pack.GetLowerBoundHost(0), pack.GetUpperBoundHost(0), kb.s, kb.e, jb.s, jb.e, ib.s,
-      ib.e,
-      KOKKOS_LAMBDA(const int n, const int k, const int j, const int i,
-                    typename Kokkos::MinMax<Real>::value_type &lminmax) {
-        lminmax.min_val =
-            (pack(n, k, j, i) < lminmax.min_val ? pack(n, k, j, i) : lminmax.min_val);
-        lminmax.max_val =
-            (pack(n, k, j, i) > lminmax.max_val ? pack(n, k, j, i) : lminmax.max_val);
-      },
-      Kokkos::MinMax<Real>(minmax));
+    typename Kokkos::MinMax<Real>::value_type minmax;
+    parthenon::par_reduce(
+        parthenon::loop_pattern_mdrange_tag, PARTHENON_AUTO_LABEL, DevExecSpace(), 0,
+        pack.GetNBlocks() - 1, // Runs from [0, 0] since pack built from MeshBlockData
```

Comment:
Sparse packs always require a block index to disambiguate flat pack calls. Of course, for single block packs it is not necessary to explicitly include the block index in the loop but I usually include it so that if a kernel gets switched to being over `MeshData` instead of `MeshBlockData` the loops don't need to be changed.